engine/src/main/scala/cromwell/webservice/PartialWorkflowSources.scala

```diff
@@ -195,14 +205,31 @@ object PartialWorkflowSources {
       }
     }
 
+    def validateWorkflowUrl(workflowUrl: Option[String]): ErrorOr[Option[String]] = {
+      def convertUrlToUri(workflowUrl: String): ErrorOr[Option[String]] = {
```

Comment:
ToL: This doesn't really seem to convert a Url to a Uri but rather validate that a String is a valid Uri.
Maybe the function name could be something like `validateStringAsUri` ?

Also you might be able to simplify it a little by returning an `ErrorOr[String]` (removing the Option) and simply doing

```scala
import cats.syntax.traverse._
import cats.instances.option._
workflowUrl.traverse(validateStringAsUri)
```

Reply:
It's nice to express the fact that something has been validated in the final type (eg should this return `Url`? or maybe there's a `String Refined Url`?

Reply:
@cjllanwarne I have updated it to return `ErrorOr[WorkflowUrl]` instead of `ErrorOr[String]`.